core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -158,6 +161,17 @@ public LogicalPlan visitRelationSubquery(RelationSubquery node, AnalysisContext
     return subquery;
   }
 
+  @Override
+  public LogicalPlan visitTableFunction(TableFunction node, AnalysisContext context) {
+    Function function = (Function) node.getTableFunction();
+    String catalogName
+        = getCatalogNameFromQualifiedTableFunctionName(function.getFuncName());
+    Expression tableFunction = expressionAnalyzer.analyze(function, context);
+    return new LogicalTableFunction(tableFunction,
```

Comment:
* If tableFunction is bind to storageEngine. should we return LogicalRelation instead of LogicalTableFunction. e.g.

```
Table table = catalogService.getStorageEngine(catalogName).getTableFunction(tableFunctionName);
return new LogicalRelation(tableFunctionName, table)
```

Reply:
What Peng suggested is similar as Calcite Schema.getFunction(name) which our `StorageEngine` interface references: https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/schema/Schema.java#L96